Show appropriate messages on Registration Confirmation link failure

arslanashraf7 opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#1885 

#### What's this PR do?
It add proper messages for more user readability on validation fail. Adds two more states to support the error messages:

1. If the user already has an account/partial account created and he clicks the confirmation link again he will be shown a message to login again. (`You already have an xPRO account. Please [click here] to sign in.`)
2. If a user doesn't have an account and he clicks on the confirmation link again he shown invalid/expired link message. (`This invitation is invalid or has expired. Please [click here] to register again.`)

#### How should this be manually tested?
Testing instruction are added in the associated ticket.

#### Where should the reviewer start?
Account Registration

#### Screenshots (if appropriate)
**Screenshot when User has partial/full profile created with us -- 'Click Here' Link will take him to login page**
<img width="1440" alt="Screenshot 2021-02-11 at 12 28 15 PM" src="https://user-images.githubusercontent.com/34372316/107615155-7c71cd00-6c6d-11eb-9c20-f41fb112dca2.png">

**Screenshot when User doesn't have a profile created with us -- 'Click Here' Link will take him to register page**

<img width="1439" alt="Screenshot 2021-02-11 at 12 26 48 PM" src="https://user-images.githubusercontent.com/34372316/107615128-70860b00-6c6d-11eb-8d79-8b58182984ef.png">

asadiqbal08 reviewed (approved):
👍

arslanashraf7 commented:
@briangrossman Would you like to have a look on the screenshots, Its approved by Asad and it would be great of you could review the designs and scenarios mentioned in **What's this PR do** section.

briangrossman commented:
@arslanashraf7 Please use a lowercase `c`, rather than upper case `C` for `click here` in both cases. Otherwise, looks great.

arslanashraf7 commented:
> @arslanashraf7 Please use a lowercase `c`, rather than upper case `C` for `click here` in both cases. Also, please extend the link to include the words `to sign in` and `to register again` in the link.
> 
> `You already have an xPRO account. Please [click here to sign in].`
> 
> Otherwise, looks great.

Thanks Brian for the review, I've incorporated text/link changes and updated the screenshots in the PR description in case you would like to have a look.

briangrossman commented:
@arslanashraf7 I just tested on RC and it's working well.
